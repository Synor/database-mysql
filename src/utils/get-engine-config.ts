import { SynorError } from '@synor/core'

type MySQLDatabaseEngineConfig = {
  migrationTableName: string
}

export function getEngineConfig(uri: string): MySQLDatabaseEngineConfig {
  const { protocol, searchParams: params } = new URL(uri)

  if (protocol !== 'mysql:') {
    throw new SynorError(`Invalid DatabaseURI`)
  }

  const migrationTableName =
    params.get('synor-migration-record-table') || 'synor_migration_record'

  return {
    migrationTableName
  }
}
