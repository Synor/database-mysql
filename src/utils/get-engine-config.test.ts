import { getEngineConfig } from './get-engine-config'

describe('utils:getEngineConfig', () => {
  let uri: Parameters<typeof getEngineConfig>[0]

  beforeEach(() => {
    uri = 'mysql://localhost/synor'
  })

  test('accepts mysql uri', () => {
    expect(getEngineConfig(uri)).toMatchSnapshot()
  })

  test('accepts custom migration record table name', () => {
    const tableName = 'migration_history'
    uri = `${uri}?synor-migration-record-table=${tableName}`
    expect(getEngineConfig(uri).migrationTableName).toBe(tableName)
  })

  test(`throws if uri protocol is not 'mysql:'`, () => {
    uri = 'postgresql://localhost/synor'
    expect(() => getEngineConfig(uri)).toThrow()
  })

  test('throws if uri is malformed', () => {
    uri = 'mysql://@_@/synor'
    expect(() => getEngineConfig(uri)).toThrow()
  })
})
