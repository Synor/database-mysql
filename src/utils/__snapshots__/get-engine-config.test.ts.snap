// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils:getEngineConfig accepts mysql uri 1`] = `
Object {
  "migrationTableName": "synor_migration_record",
}
`;
